code/modules/projectiles/mounted.dm

```diff
@@ -9,13 +9,20 @@
 	hud_possible = list(MACHINE_HEALTH_HUD, SENTRY_AMMO_HUD)
 
 ///generates the icon based on how much ammo it has.
-/obj/machinery/deployable/mounted/update_icon_state(mob/user)
+/obj/machinery/deployable/mounted/update_icon(mob/user)
 	. = ..()
 	var/obj/item/weapon/gun/gun = internal_item
-	if(!gun.current_mag)
+	var/has_mag
+	if(istype(gun, /obj/item/weapon/gun/energy))
+		var/obj/item/weapon/gun/energy/energy_gun = gun
+		has_mag = energy_gun.cell
+	else
+		has_mag = gun.current_mag
+	if(!has_mag)
 		icon_state = default_icon_state + "_e"
 	else
 		icon_state = default_icon_state
+
```

Comment:
Why change this still looks like icon state only updates to me

Reply:
Still update icon state yes